pkgs/applications/editors/vscode/generic.nix

```diff
@@ -16,6 +16,9 @@
 # sourceExecutableName is the name of the binary in the source archive, over
 # which we have no control
 , sourceExecutableName ? executableName
+
+, useBundledRipgrep ? false
```

Comment:
I think this would be better set to `true` by default, since using an unbundled RipGrep can probably cause some issues.

Reply:
After having a look at Microsoft's `ripgrep` build scripts, I'm still leaning toward setting this to `false` by default as they just include PCRE2 by default (just like our Nix package) https://github.com/microsoft/ripgrep-prebuilt/blob/main/build/build.sh#L28 and I would rather use one less bundled executable if possible.

Reply:
I would still prefer this being setting to `true` by default. My main concern here is difference in ripgrep versions causing issues or invaliding bug reports to upstream.

Folks who need this flag (like your case) will have to explicitly opt-in, and in this case they will probably disable this flag first before reporting issues upstream.

Reply:
In general the NIxpkgs policy is always to unbundle bundled dependencies AFAIK.

Reply:
I don't remember Nixpkgs having a explicitly policy for this. When it makes sense it is better to unbundle, but I am not sure in this case.

I am be wrong though.

Reply:
nit: could we also change the name `useBundledRipgrep`? it wasn't immediately clear to me what "bundled" meant in this context. What about something more explicit like `useNixpkgsRipgrep` or `useVSCodeRipgrep`?

Reply:
Changed to `useVSCodeRipgrep` :)

Reply:
> invaliding bug reports to upstream.

I think thats already the case tbh.

> My main concern here is difference in ripgrep versions causing issues

Yes. It would be very annoying if a ripgrep update was blocked by vscode